platforms/Linux/DEB/Ubuntu/shared/debian/rules

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/make -f
+# See debhelper(7) (uncomment to enable)
+# output every command that modifies files on the build system.
+#export DH_VERBOSE = 1
+
+BUILD_SCRIPT_ARGS =
+ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+    NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+    BUILD_SCRIPT_ARGS += -j $(NUMJOBS)
+endif
+
+# Verbose Swift build
+export VERBOSE=1
+
+SWIFTLANG_PKG_VER=$(shell dpkg-parsechangelog -S Version | sed -e 's/-[^-]*$$//')
+SWIFT_BUILDDIR=$(CURDIR)/build
+
+
+export DEB_BUILD_MAINT_OPTIONS=hardening=+format,+fortify,+stackprotectorstrong,+relro,optimize=-lto
+export MOD_CXXFLAGS=$(DEB_CXXFLAGS_STRIP="-ffat-lto-objects -flto=auto -flto=auto" dpkg-buildflags --get CXXFLAGS)
+export MOD_CFLAGS=$(DEB_CFLAGS_STRIP="-ffat-lto-objects -flto=auto -flto=auto" dpkg-buildflags --get CFLAGS)
+export MOD_LDFLAGS=$(DEB_LDFLAGS_STRIP="-ffat-lto-objects -flto=auto -flto=auto" dpkg-buildflags --get LDFLAGS)
+
+%:
+	dh $@
+
+clean::
+	dh_testdir
+	
+	# Generate debian/control and others
+	sed -e 's/x.y.z/$(SWIFTLANG_PKG_VER)/g' < debian/control.in > debian/control
+	sed -e 's/X.Y.Z/$(SWIFTLANG_PKG_VER)/g' < debian/swiftlang.install.in > debian/swiftlang-$(SWIFTLANG_PKG_VER)-bin.install
+	sed -e 's/X.Y.Z/$(SWIFTLANG_PKG_VER)/g' < debian/libswiftlang.install.in > debian/libswiftlang-$(SWIFTLANG_PKG_VER).install
+	sed -e 's/X.Y.Z/$(SWIFTLANG_PKG_VER)/g' < debian/libswiftlang-dev.install.in > debian/libswiftlang-$(SWIFTLANG_PKG_VER)-dev.install
+	sed -e 's/X.Y.Z/$(SWIFTLANG_PKG_VER)/g' < debian/swiftlang.postinst.in > debian/swiftlang-$(SWIFTLANG_PKG_VER)-bin.postinst
+	sed -e 's/X.Y.Z/$(SWIFTLANG_PKG_VER)/g' < debian/swiftlang.prerm.in > debian/swiftlang-$(SWIFTLANG_PKG_VER)-bin.prerm
+
+	chmod ugo+x debian/swiftlang-$(SWIFTLANG_PKG_VER)-bin.postinst
+	chmod ugo+x debian/swiftlang-$(SWIFTLANG_PKG_VER)-bin.prerm
+
+	rm -rf $(SWIFT_BUILDDIR)
+	
+	# Modified during the build
+	rm -f swift-corelibs-libdispatch/dispatch/module.modulemap
+	rm -f swift-corelibs-libdispatch/private/module.modulemap
+	
+	# Python byte-compiled files
+	find swift/utils -name '*.pyc' -delete
+	find llvm-project -name '*.pyc' -delete
+	find cmark/test -name '*.pyc' -delete
+	find swiftpm -name '*.pyc' -delete
+	
+	dh_clean
+
+override_dh_auto_clean:
+	# Disabled; we have a non-standard build system that the tool
+	# doesn't support, so be on the safe side and avoid any potential
+	# false-triggering.
+
+override_dh_update_autotools_config:
+	# Don't go and update config.guess/config.sub, it causes LLVM to
+	# pick up the x86_64-pc-linux-gnu triplet instead of the expected
+	# x86_64-unknown-linux-gnu. This causes Swift modules to break.
+
+override_dh_autoreconf:
+	# We'd rather not do that either, see above.
+
+override_dh_shlibdeps:
+	dh_shlibdeps -- -xlibswiftlang-$(SWIFTLANG_PKG_VER) -xswiftlang-$(SWIFTLANG_PKG_VER)-bin
```

Comment:
The division into `libswiftlang-X.Y.Z` and `libswiftlang-X.Y.Z-bin` necessitated:

- removing the `override_dh_makeshlibs` from above (it was made a no-op previously).
- these `-x` params passed to `dh_slibdeps`. In their absence, including `${shlibs:Depends}` for these packages in the control file would result in self-references in the `Depends` lists for these packages (they would list themselves as a dependency). I guess because there are shared libraries that depend on other shared libraries within these packages, and  `dh_makeshlibs` when run mostly fails to list shared library names and versions since `SOVERSION` is mostly not set anywhere in the Swift libraries built. This is a guess, functionally things seem OK, but I figured further cleanup on this topic would be follow-up material.

Re: making `dh_makeshlibs` actually produce its output: with the packages divided into a lib and bin package, I did not manage to find flags which would make `dh_shlibdeps` stage pass without `dh_makeshlibs` also being run (an alternative would I guess be a `shlibs.local` file, which gets quite tedious too?)